llvm/tools/repo2obj/R2OELFSymbolTable.h

```diff
@@ -312,8 +316,10 @@ auto SymbolTable<ELFT>::sort() -> std::vector<Value *> {
   std::stable_sort(
       std::begin(OrderedSymbols), std::end(OrderedSymbols),
       [](Value const *const A, Value const *const B) {
-        return A->linkage() == pstore::repo::linkage_type::internal &&
-               B->linkage() != pstore::repo::linkage_type::internal;
+        return (A->linkage() == pstore::repo::linkage_type::internal ||
+                A->linkage() == pstore::repo::linkage_type::internal_no_symbol) &&
```

Comment:
Better not to call `Value::linkage()` twice (it has logic to handle the def/undef case).

Reply:
Thanks, Fixed it.